src/MasterBlueprintServiceManager.sol

```diff
@@ -0,0 +1,482 @@
+
+    /// @dev Mapping that store the blueprint service manager contracts.
+    /// @notice Contains the blueprints created by the runtime.
+    ///
+    /// blueprintId => Blueprint Service Manager address
+    EnumerableMap.UintToAddressMap private blueprints;
+
+    /// @dev Mapping to store the blueprint owners.
+    /// @notice Contains the owner of the blueprints.
+    ///
+    /// blueprintId => owner
+    EnumerableMap.UintToAddressMap private blueprintOwners;
+
+    // ========= Errors =========
+
+    // ========== Modifiers ==========
+
+    // ======== Functions =========
+
+    /**
+     * @dev Hook to handle blueprint creation. Gets called by the root chain when a new blueprint is created.
+     * @param blueprintId The unique identifier for the blueprint.
+     * @param owner The address of the blueprint owner.
+     * @param blueprint The blueprint data.
+     */
+    function onBlueprintCreated(
+        uint64 blueprintId,
+        address owner,
+        Blueprint calldata blueprint
+    )
+        public
+        payable
+        onlyFromRootChain
+        whenNotPaused
+    {
+        blueprints.set(blueprintId, blueprint.manager);
+        blueprintOwners.set(blueprintId, owner);
+        emit BlueprintCreated(owner, blueprintId, blueprint);
+    }
+
+    /**
+     * @dev Called by the runtime when a service operator attempts to register with the blueprint.
+     * @param blueprintId The blueprint unique identifier.
+     * @param operator The Service Operator.
+     * @param registrationInputs The registration inputs in bytes format.
+     */
+    function onRegister(
+        uint64 blueprintId,
+        ServiceOperators.OperatorPreferences calldata operator,
+        bytes calldata registrationInputs
+    )
+        public
+        payable
+        onlyFromRootChain
+        whenNotPaused
+    {
+        address manager = blueprints.get(blueprintId);
+        // TODO: Implement additional registration logic if needed.
+        IBlueprintServiceManager(manager).onRegister(operator, registrationInputs);
+        emit OperatorRegistered(blueprintId, operator);
+    }
+
+    /**
+     * @dev Called when a service operator attempts to unregister from the blueprint.
+     * @param blueprintId The blueprint unique identifier.
+     * @param operator The operator's details.
+     */
+    function onUnregister(
+        uint64 blueprintId,
+        ServiceOperators.OperatorPreferences calldata operator
+    )
+        public
+        onlyFromRootChain
+        whenNotPaused
+    {
+        address manager = blueprints.get(blueprintId);
+        IBlueprintServiceManager(manager).onUnregister(operator);
+        emit OperatorUnregistered(blueprintId, operator);
+    }
+
+    /**
+     * @dev Called when an operator updates their price targets.
+     * @param blueprintId The blueprint unique identifier.
+     * @param operator The operator's details with the updated price targets.
+     */
+    function onUpdatePriceTargets(
```

Comment:
This hook streams all the updates by operators who are updating their price targets for blueprint specified by `blueprintId`. How it will effect live instances (already running service instances from this blueprint, I assume) is up to the `BlueprintServiceManager` implementation.